runtime/shared/shrclssup.c

```diff
@@ -178,10 +178,18 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 					memcpy(vm->sharedCacheAPI->methodSpecs, methodSpecs, strlen(methodSpecs) + 1);
 				}
 
+				if (runtimeFlags & J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE) {
+					vm->sharedCacheAPI->cacheType = J9PORT_SHR_CACHE_TYPE_PERSISTENT;
+				} else {
+					vm->sharedCacheAPI->cacheType = J9PORT_SHR_CACHE_TYPE_NONPERSISTENT;
+				}
+				/* set runtimeFlags and verboseFlags here as they will be used later in SH_OSCache::getCacheDir() */
+				vm->sharedCacheAPI->runtimeFlags = runtimeFlags;
+				vm->sharedCacheAPI->verboseFlags = verboseFlags;
+
 #if !defined(WIN32) && !defined(WIN64)
 				/* Get platform default cache directory */
-				rc = j9shmem_getDir(NULL, J9SHMEM_GETDIR_APPEND_BASEDIR, defaultCacheDir, J9SH_MAXPATH);
-				if (-1 == rc) {
+				if (-1 == j9shr_getCacheDir(vm, NULL, defaultCacheDir, J9SH_MAXPATH, vm->sharedCacheAPI->cacheType)) {
```

Comment:
This condition will always be false.
>  * @internal J9PORT_ERROR_SHMEM* range from at -170 to -199 to match hyporterror.h codes.

Reply:
I think that is intentional? If it did happen we'd hit the assert.

Reply:
j9shr_getCacheDir() and SH_OSCache::getCacheDir() returns -1 on failure. I think the assert needs to be removed.

Reply:
I will remove this assert in the next PR.

Reply:
PR: https://github.com/eclipse/openj9/pull/5202

Reply:
You found a problem anyway :)